tests/python_functional/src/syslog_ng_config/syslog_ng_config.py

```diff
@@ -108,3 +109,10 @@ def create_logpath(self, statements=None, flags=None):
     def create_inner_logpath(self, statements=None, flags=None):
         inner_logpath = self.__create_logpath_with_conversion(statements, flags)
         return inner_logpath
+
+    def create_generator_source(self, freq=1, num=0, template=None):
```

Comment:
1, About the config-create function names (in general), my preference is: "create+current_driver_name()" -> `create_example_msg_generator()` would be better here. Be consequent with our driver names. 
But...
2, I like the approach to create drivers with default options (I am also thinking about this in one of my next PRs),
but I would not dismiss the possibility to create drivers with any options. So I would do the following changes here:
a, rename the current function to: `create_basic_example_msg_generator()` -> I would define the basic default options from inside the function (as you did it from parameters).
b, create a new function like: `create_example_msg_generator(**kwargs)` -> Further options can be passed via **kwargs

Reply:
Sure thing. I will do so.